apps/voting/app/src/script.js

```diff
@@ -223,17 +223,24 @@ async function loadVoteDescription(vote) {
     return vote
   }
 
-  const path = await app.describeScript(vote.script).toPromise()
-  vote.description = path
-    ? path
-        .map(step => {
-          const identifier = step.identifier ? ` (${step.identifier})` : ''
-          const app = step.name ? `${step.name}${identifier}` : `${step.to}`
-
-          return `${app}: ${step.description || 'No description'}`
-        })
-        .join('\n')
-    : ''
+  try {
+    const path = await app.describeScript(vote.script).toPromise()
+
+    vote.description = path
+      ? path
+          .map(step => {
+            const identifier = step.identifier ? ` (${step.identifier})` : ''
+            const app = step.name ? `${step.name}${identifier}` : `${step.to}`
+
+            return `${app}: ${step.description || 'No description'}`
+          })
+          .join('\n')
+      : ''
+  } catch (error) {
+    console.log('Error describing vote script', error)
+    vote.description = 'Invalid script. The result cannot be executed.'
+    vote.metadata = ''
```

Comment:
Why is this necessary?

Reply:
That's how actions are differentiated from question votes. It's not really clear here, but if you look at the reduced state, all votes with a script have empty metadata. Because this fails, we have to manually set it to an empty string. Otherwise it's rendered as a question vote.

Reply:
https://github.com/aragon/aragon-apps/blob/catch-invalid-script-error/apps/voting/app/src/components/VotePanel.js#L114-L117

Reply:
Sorry, about my lack of context, in the code before there was no setting of metadata but now there is, I was wondering what it was for. Thanks for clarifying.

Reply:
From the link you share, maybe setting the value to `null` would be better no?

Reply:
What would happen now if a vote has both description (from a script) and metadata? The description should always take precedence

Reply:
In the vote panel they're each rendered if truthy. 

In the card metadata takes precedence:
https://github.com/aragon/aragon-apps/blob/51e480001c6975baf0721164a07f8815e973b97c/apps/voting/app/src/components/VotingCard/VotingCard.js#L79

Should we switch the two?

Reply:
We should, yes